### PR TITLE
Rcal 727 Add High Level Pipeline documentation, various updates

### LINKED ISSUE #1034: Add a section for the HLP to RTD
_Issue [RCAL-727](https://jira.stsci.edu/browse/RCAL-727) was created on JIRA by [David Davis](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ddavis):_

Add a description of the HLP to the roman RTD ([https://roman-pipeline.readthedocs.io/en/latest/roman/pipeline_installation.html)]

### PR DESCRIPTION
**Pull request by ddavis-stsci**
Resolves [RCAL-727](https://jira.stsci.edu/browse/RCAL-727)

Closes #1034 

This PR updates the documentation for the 24Q1_B12 release. 
Section for HLP added
Update ELP for combined Jump/Ramp fitting
Added source detection and tweakreg to the steps and tables

**Checklist**
- [N/A] added entry in `CHANGES.rst` under the corresponding subsection
- [N/A] updated relevant tests
- [x] updated relevant documentation
- [x] updated relevant milestone(s)
- [x] added relevant label(s)
- [N/A] ran regression tests, post a link to the Jenkins job below. [How to run regression tests on a PR](https://github.com/spacetelescope/romancal/wiki/Running-Regression-Tests-Against-PR-Branches)

### CODE REVIEW

File: docs/roman/pipeline/main.rst [hunk 1]
```diff
@@ -3,37 +3,42 @@
...
 ===============
 
-End-to-end calibration of Roman data is divided into stages of
+End-to-end calibration of Roman data is divided into levels of
 processing:
 
-- Stage 2 consists of detector-level corrections that are performed on a
-  group-by-group basis, followed by ramp fitting. The output of stage 1
+- The exposure level processing consists of detector-level corrections that are performed on a
+  group-by-group basis, followed by ramp fitting. The output of the esposure level
```
Comment (schlafly) [inline docs/roman/pipeline/main.rst:10]:
```suggestion
  group-by-group basis, followed by ramp fitting. The output of the exposure level
```

File: docs/roman/pipeline/main.rst [hunk 2]
```diff
@@ -3,37 +3,42 @@
...
   processing is a count rate image per exposure, or per integration for
-  some modes. Details of this pipeline can be found at:
+  some modes that is aligned to the GAIA reference system.
```
Comment (schlafly) [inline docs/roman/pipeline/main.rst:12]:
```suggestion
  some modes that is aligned to the Gaia reference system.
```

**Review by schlafly — APPROVED** [review]
Looks good to me.  I left a few minor changes in line.

File: docs/roman/pipeline/main.rst (hunk 2 above)
Comment (ddavis-stsci) [reply docs/roman/pipeline/main.rst]: Done

**Pull request merged**